### PR TITLE
Freshen up documentation for Document JSON updates

### PR DESCRIPTION
**Pull request by vekterli**
@geirst @kkraune please review

Restructures doc to be oriented more around the type of field being
updated, since that's likely to be what people are looking for.

### CODE REVIEW

File: documentation/reference/document-json-update-format.html [hunk 1]
```diff
@@ -74,90 +89,205 @@ <h3 id="assign">assign</h3>
...
+  field last_name type string {}
+}
+field contact type person {
+  indexing: summary
+}
+</pre>
 <pre>
 {
-    "update": "id:music:music::example2",
+    "update": "id:mynamespace:workers::example",
```
Comment (kkraune) [inline documentation/reference/document-json-update-format.html]: yes! good practice to say mynamespace wherever, users are confused about music:music :-)

Comment (vekterli) [reply documentation/reference/document-json-update-format.html]: I still get confused myself, to be honest 😬 Principle of Least Surprise™️ reigns supreme

File: documentation/reference/document-json-update-format.html [hunk 2]
```diff
@@ -74,90 +89,205 @@ <h3 id="assign">assign</h3>
...
+<p>
+<strong>Note:</strong> this does not work for <em>attribute</em> struct fields.
+</p>
+
+<h3 id="assign-map-field">Map field</h3>
+<p>
+Individual map entries may be updated using <a href="document-field-path.html">field path</a> syntax.
+</p>
+<p>
+Assigning an element to a key in a map will insert the key/value mapping if does not already exist, or
```
Comment (geirst) [inline documentation/reference/document-json-update-format.html]: Typo? "...mapping if does not..." -> "...mapping if it does not"?

Comment (vekterli) [reply documentation/reference/document-json-update-format.html]: Fixed

**Pull request merged**